modules/ref_fb_module/tests/test_fb_trigger.cpp

```diff
@@ -0,0 +1,329 @@
+                                    mockPackets,
+                                    thresholdChangesAfterPackets,
+                                    {},
+                                    newThresholds);
+}
+
+TEST_F(TriggerTest, TriggerTestIntExplicit)
+{
+    vecvec<Int> mockPackets{{0, 0, 0, 289, 289, 289, 289, 0, 0, 0, 289, 289, 289}, {289, 0, 0, 289, 289}, {289, 289, 0}, {0, 0, 289, 289}};
+    vecvec<Int> mockDomainPackets{{3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27}, {29, 31, 33, 35, 37}, {39, 41, 43}, {45, 47, 49, 51}};
+    vecvec<Bool> expectedData{{true, false, true}, {false, true}, {false}, {true}};
+    vecvec<Int> expectedDomain{{9, 17, 23}, {31, 35}, {43}, {49}};
+
+    auto helper = TriggerTestHelper(
+        ExplicitDataRule(), expectedData, expectedDomain, SampleTypeFromType<Int>().SampleType, mockPackets, {}, mockDomainPackets);
+}
+
+TEST_F(TriggerTest, TriggerTestIntExplicitThresholdChanged)
+{
+    vecvec<Int> mockPackets{{0, 0, 0, 289, 289, 289, 289, 0, 0, 0, 289, 289, 289}, {289, 0, 0, 289, 289}, {289, 289, 0}, {0, 0, 289, 289}};
+    vecvec<Int> mockDomainPackets{{3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27}, {29, 31, 33, 35, 37}, {39, 41, 43}, {45, 47, 49, 51}};
+    vecvec<Bool> expectedData{{true, false, true}, {false, true}, {false}, {true}};
+    vecvec<Int> expectedDomain{{9, 17, 23}, {31, 35}, {43}, {49}};
+    std::vector<Int> thresholdChangesAfterPackets{2};
+    std::vector<Float> newThresholds{200};
+
+    auto helper = TriggerTestHelper(ExplicitDataRule(),
+                                    expectedData,
+                                    expectedDomain,
+                                    SampleTypeFromType<Int>().SampleType,
+                                    mockPackets,
+                                    thresholdChangesAfterPackets,
+                                    mockDomainPackets,
+                                    newThresholds);
+}
+
+TEST_F(TriggerTest, TriggerTestIntLinear)
+{
+    vecvec<Int> mockPackets{{0, 0, 0, 289, 289, 289, 289, 0, 0, 0, 289, 289, 289}, {289, 0, 0, 289, 289}, {289, 289, 0}, {0, 0, 289, 289}};
+    vecvec<Bool> expectedData{{true, false, true}, {false, true}, {false}, {true}};
+    vecvec<Int> expectedDomain{{9, 17, 23}, {31, 35}, {43}, {49}};
+
+    auto helper =
+        TriggerTestHelper(LinearDataRule(2, 3), expectedData, expectedDomain, SampleTypeFromType<Int>().SampleType, mockPackets, {});
+}
+
+TEST_F(TriggerTest, TriggerTestIntLinearThresholdChanged)
+{
+    vecvec<Int> mockPackets{
+        {0, 0, 0, 289, 289, 289, 289, 0, 0, 0, 289, 289, 289}, {289, 0, 0, 289, 289}, {289, 289, 0}, {199, 201, 199, 289}};
+    vecvec<Bool> expectedData{{true, false, true}, {false, true}, {false}, {true, false}};
+    vecvec<Int> expectedDomain{{9, 17, 23}, {31, 35}, {43}, {47, 49}};
+    std::vector<Int> thresholdChangesAfterPackets{2};
+    std::vector<Float> newThresholds{200};
+
+    auto helper = TriggerTestHelper(LinearDataRule(2, 3),
```

Comment:
This is not entirely clear on what exactly happens. There's a new class assigned to a variable, but there's no indication of what exactly happens.

I'd move the `run()` method out of the constructor and call it explicitly.